components/brave_wallet/browser/brave_wallet_service.cc

```diff
@@ -74,6 +75,14 @@ brave_wallet::SwapController* BraveWalletService::swap_controller() const {
   return swap_controller_.get();
 }
 
+std::string BraveWalletService::WalletAccountNames() const {
+  return prefs_->GetString(kBraveWalletAccountNames);
+}
+
+void BraveWalletService::UpdateAccountNames(const std::string& accountNames) {
+  prefs_->SetString(kBraveWalletAccountNames, accountNames);
```

Comment:
cc @darkdh for a suggestion on how best to persist this. Maybe a string of account names should be passed in differently and then serialized to / from JSON?

Reply:
Please use `std::vector<base::Value>` with `STRING` type so it would match underlying account vector in `HDKeyring` with the same index. And use `PrefService::Set` to set that list value